services/alertmanager/alertmanager.go

```diff
@@ -0,0 +1,233 @@
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package alertmanager contains business logic of working with Alertmanager.
+package alertmanager
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/AlekSi/pointer"
+	httptransport "github.com/go-openapi/runtime/client"
+	"github.com/percona/pmm/api/alertmanager/amclient"
+	"github.com/percona/pmm/api/alertmanager/amclient/alert"
+	"github.com/percona/pmm/api/alertmanager/amclient/general"
+	"github.com/percona/pmm/version"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+const resendInterval = 30 * time.Second
+
+// FIXME remove completely before release
+const (
+	addTestingAlerts   = false
+	testingAlertsDelay = time.Second
+)
+
+// Service is responsible for interactions with Prometheus.
+type Service struct {
+	db             *reform.DB
+	serverVersion  *version.Parsed
+	agentsRegistry agentsRegistry
+	r              *registry
+	l              *logrus.Entry
+}
+
+// New creates new service.
+func New(db *reform.DB, v string, agentsRegistry agentsRegistry) (*Service, error) {
+	serverVersion, err := version.Parse(v)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Service{
+		db:             db,
+		serverVersion:  serverVersion,
+		agentsRegistry: agentsRegistry,
+		r:              newRegistry(),
+		l:              logrus.WithField("component", "alertmanager"),
+	}, nil
+}
+
+func (svc *Service) AddAlert(id string, delayFor time.Duration, params *AlertParams) error {
+	alert, err := makeAlert(params)
+	if err != nil {
+		return err
+	}
+
+	svc.r.Add(id, delayFor, alert)
+	return nil
+}
+
+func (svc *Service) RemoveAlert(id string) {
+	svc.r.Remove(id)
+}
+
+// Run runs Alertmanager configuration update loop until ctx is canceled.
+func (svc *Service) Run(ctx context.Context) {
+	svc.l.Info("Starting...")
+	defer svc.l.Info("Done.")
+
+	generateBaseConfig()
+
+	t := time.NewTicker(resendInterval)
+	defer t.Stop()
+
+	for {
+		if addTestingAlerts {
+			svc.updateInventoryAlerts(ctx)
+		}
+
+		svc.sendAlerts(ctx)
+
+		select {
+		case <-ctx.Done():
+			return
+		case <-t.C:
+			// nothing, continue for loop
+		}
+	}
+}
+
+// generateBaseConfig generates /srv/alertmanager/alertmanager.base.yml if it is not present.
+//
+// TODO That's a temporary measure until we start generating /etc/alertmanager.yml
+// using /srv/alertmanager/alertmanager.base.yml as a base. See supervisord config.
+func generateBaseConfig() {
```

Comment:
That file will be created by user like prometheus.base.yml: https://www.percona.com/blog/2020/03/23/extending-pmm-prometheus-configuration/ We can't do that in packaging because that file should not be a part of the package.